{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Populating the interactive namespace from numpy and matplotlib\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrX_mean_depth</th>\n",
       "      <th>pred_gender</th>\n",
       "      <th>gender</th>\n",
       "      <th>consistent</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MMI001</th>\n",
       "      <td>19.322773</td>\n",
       "      <td>M</td>\n",
       "      <td>M</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MMI002</th>\n",
       "      <td>37.040048</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MMI003</th>\n",
       "      <td>23.854752</td>\n",
       "      <td>M</td>\n",
       "      <td>M</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MMI004</th>\n",
       "      <td>46.977884</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MMI005</th>\n",
       "      <td>30.219566</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        chrX_mean_depth pred_gender gender  consistent\n",
       "MMI001        19.322773           M      M        True\n",
       "MMI002        37.040048           F      F        True\n",
       "MMI003        23.854752           M      M        True\n",
       "MMI004        46.977884           F      F        True\n",
       "MMI005        30.219566           F      F        True"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "%pylab inline\n",
    "\n",
    "df = pd.read_csv('data/MMI_gender_prediction.csv', index_col=0)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MMI011\n",
      "INFO:root:Processing sample MMI011.realigned at 2022-03-02 00:41:14.997018\n",
      "INFO:root:Writing to json at 2022-03-02 00:41:52.906988\n",
      "INFO:root:Writing to tsv at 2022-03-02 00:41:52.909281\n",
      "Call Cyrius to out/MMI011/cyrius_MMI011 done!\n",
      "\u001b[37müêø  Aldy v3.3 (Python 3.7.9 on Linux 4.18.0-305.3.1.el8.x86_64-x86_64-with-centos-8.3.2011)\u001b[39;49;00m\n",
      "\u001b[37m   (c) 2016-2022 Aldy Authors. All rights reserved.\n",
      "   Free for non-commercial/academic use only.\u001b[39;49;00m\n",
      "\u001b[37mGenotyping sample MMI011.realigned.bam...\u001b[39;49;00m\n",
      "\u001b[37mPotential CFTR gene structures for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mPotential major CFTR star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*normal (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mBest CFTR star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: *normal / *normal (confidence=100%)\u001b[39;49;00m\n",
      "\u001b[37m      Minor alleles: *normal, *normal\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01mCFTR results:\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01m  - *normal / *normal\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m    Minor: [*normal] / [*normal]\u001b[39;49;00m\n",
      "\u001b[37m    Legacy notation: [*normal] / [*normal]\u001b[39;49;00m\n",
      "\u001b[37müêø  Aldy v3.3 (Python 3.7.9 on Linux 4.18.0-305.3.1.el8.x86_64-x86_64-with-centos-8.3.2011)\u001b[39;49;00m\n",
      "\u001b[37m   (c) 2016-2022 Aldy Authors. All rights reserved.\n",
      "   Free for non-commercial/academic use only.\u001b[39;49;00m\n",
      "\u001b[37mGenotyping sample MMI011.realigned.bam...\u001b[39;49;00m\n",
      "\u001b[37mPotential CYP2B6 gene structures for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mPotential major CYP2B6 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 1x*4, 1x*6 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mBest CYP2B6 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: *4 / *6 (confidence=100%)\u001b[39;49;00m\n",
      "\u001b[37m      Minor alleles: *4.002 +rs7254579, *6.003\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01mCYP2B6 results:\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01m  - *4 / *6\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m    Minor: [*4.002 +rs7254579] / [*6.003]\u001b[39;49;00m\n",
      "\u001b[37m    Legacy notation: [*4D +rs7254579] / [*6C]\u001b[39;49;00m\n",
      "\u001b[37müêø  Aldy v3.3 (Python 3.7.9 on Linux 4.18.0-305.3.1.el8.x86_64-x86_64-with-centos-8.3.2011)\u001b[39;49;00m\n",
      "\u001b[37m   (c) 2016-2022 Aldy Authors. All rights reserved.\n",
      "   Free for non-commercial/academic use only.\u001b[39;49;00m\n",
      "\u001b[37mGenotyping sample MMI011.realigned.bam...\u001b[39;49;00m\n",
      "\u001b[37mPotential CYP2C19 gene structures for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mPotential major CYP2C19 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 1x*1B, 1x*3 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mBest CYP2C19 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: *1B / *3 (confidence=100%)\u001b[39;49;00m\n",
      "\u001b[37m      Minor alleles: *1.005 +rs17885098, *3.001\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01mCYP2C19 results:\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01m  - *1B / *3\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m    Minor: [*1.005 +rs17885098] / [*3.001]\u001b[39;49;00m\n",
      "\u001b[37m    Legacy notation: [*1.005 +rs17885098] / [*3A]\u001b[39;49;00m\n",
      "\u001b[37müêø  Aldy v3.3 (Python 3.7.9 on Linux 4.18.0-305.3.1.el8.x86_64-x86_64-with-centos-8.3.2011)\u001b[39;49;00m\n",
      "\u001b[37m   (c) 2016-2022 Aldy Authors. All rights reserved.\n",
      "   Free for non-commercial/academic use only.\u001b[39;49;00m\n",
      "\u001b[37mGenotyping sample MMI011.realigned.bam...\u001b[39;49;00m\n",
      "\u001b[37mPotential CYP2C8 gene structures for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mPotential major CYP2C8 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mBest CYP2C8 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: *1 / *1 (confidence=100%)\u001b[39;49;00m\n",
      "\u001b[37m      Minor alleles: *1.001, *1.002\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01mCYP2C8 results:\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01m  - *1 / *1\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m    Minor: [*1.001] / [*1.002]\u001b[39;49;00m\n",
      "\u001b[37m    Legacy notation: [*1A] / [*1B]\u001b[39;49;00m\n",
      "\u001b[37müêø  Aldy v3.3 (Python 3.7.9 on Linux 4.18.0-305.3.1.el8.x86_64-x86_64-with-centos-8.3.2011)\u001b[39;49;00m\n",
      "\u001b[37m   (c) 2016-2022 Aldy Authors. All rights reserved.\n",
      "   Free for non-commercial/academic use only.\u001b[39;49;00m\n",
      "\u001b[37mGenotyping sample MMI011.realigned.bam...\u001b[39;49;00m\n",
      "\u001b[37mPotential CYP2C9 gene structures for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mPotential major CYP2C9 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mBest CYP2C9 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: *1 / *1 (confidence=100%)\u001b[39;49;00m\n",
      "\u001b[37m      Minor alleles: *1.001, *1.006\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01mCYP2C9 results:\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01m  - *1 / *1\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m    Minor: [*1.001] / [*1.006]\u001b[39;49;00m\n",
      "\u001b[37m    Legacy notation: [*1A] / [*1.006]\u001b[39;49;00m\n",
      "\u001b[37müêø  Aldy v3.3 (Python 3.7.9 on Linux 4.18.0-305.3.1.el8.x86_64-x86_64-with-centos-8.3.2011)\u001b[39;49;00m\n",
      "\u001b[37m   (c) 2016-2022 Aldy Authors. All rights reserved.\n",
      "   Free for non-commercial/academic use only.\u001b[39;49;00m\n",
      "\u001b[37mGenotyping sample MMI011.realigned.bam...\u001b[39;49;00m\n",
      "\u001b[37mPotential CYP2D6 gene structures for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 1x*1,1x*36.ALDY (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37m   2: 1x*1,1x*61 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mPotential major CYP2D6 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 1x*10, 1x*36.ALDY (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mBest CYP2D6 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: *10 / *36.ALDY (confidence=100%)\u001b[39;49;00m\n",
      "\u001b[37m      Minor alleles: *10.002, *36.1001\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01mCYP2D6 results:\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01m  - *10 / *36.ALDY\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m    Minor: [*10.002] / [*36.1001]\u001b[39;49;00m\n",
      "\u001b[37m    Legacy notation: [*10B] / [*36.ALDY]\u001b[39;49;00m\n",
      "\u001b[37müêø  Aldy v3.3 (Python 3.7.9 on Linux 4.18.0-305.3.1.el8.x86_64-x86_64-with-centos-8.3.2011)\u001b[39;49;00m\n",
      "\u001b[37m   (c) 2016-2022 Aldy Authors. All rights reserved.\n",
      "   Free for non-commercial/academic use only.\u001b[39;49;00m\n",
      "\u001b[37mGenotyping sample MMI011.realigned.bam...\u001b[39;49;00m\n",
      "\u001b[37mPotential CYP3A5 gene structures for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mPotential major CYP3A5 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*3 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mBest CYP3A5 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: *3 / *3 (confidence=100%)\u001b[39;49;00m\n",
      "\u001b[37m      Minor alleles: *3.001, *3.001\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01mCYP3A5 results:\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01m  - *3 / *3\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m    Minor: [*3.001] / [*3.001]\u001b[39;49;00m\n",
      "\u001b[37m    Legacy notation: [*3A] / [*3A]\u001b[39;49;00m\n",
      "\u001b[37müêø  Aldy v3.3 (Python 3.7.9 on Linux 4.18.0-305.3.1.el8.x86_64-x86_64-with-centos-8.3.2011)\u001b[39;49;00m\n",
      "\u001b[37m   (c) 2016-2022 Aldy Authors. All rights reserved.\n",
      "   Free for non-commercial/academic use only.\u001b[39;49;00m\n",
      "\u001b[37mGenotyping sample MMI011.realigned.bam...\u001b[39;49;00m\n",
      "\u001b[37mPotential CYP4F2 gene structures for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mPotential major CYP4F2 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mBest CYP4F2 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: *1 / *1 (confidence=100%)\u001b[39;49;00m\n",
      "\u001b[37m      Minor alleles: *1.001, *1.001\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01mCYP4F2 results:\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01m  - *1 / *1\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m    Minor: [*1.001] / [*1.001]\u001b[39;49;00m\n",
      "\u001b[37m    Legacy notation: [*1] / [*1]\u001b[39;49;00m\n",
      "\u001b[37müêø  Aldy v3.3 (Python 3.7.9 on Linux 4.18.0-305.3.1.el8.x86_64-x86_64-with-centos-8.3.2011)\u001b[39;49;00m\n",
      "\u001b[37m   (c) 2016-2022 Aldy Authors. All rights reserved.\n",
      "   Free for non-commercial/academic use only.\u001b[39;49;00m\n",
      "\u001b[37mGenotyping sample MMI011.realigned.bam...\u001b[39;49;00m\n",
      "\u001b[37mPotential DPYD gene structures for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mPotential major DPYD star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 1x*1, 1x*rs17376848 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mBest DPYD star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: *1 / *rs17376848 (confidence=100%)\u001b[39;49;00m\n",
      "\u001b[37m      Minor alleles: *1, *rs17376848\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01mDPYD results:\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01m  - *1 / *rs17376848\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m    Minor: [*1] / [*rs17376848]\u001b[39;49;00m\n",
      "\u001b[37m    Legacy notation: [*1] / [*rs17376848]\u001b[39;49;00m\n",
      "\u001b[37müêø  Aldy v3.3 (Python 3.7.9 on Linux 4.18.0-305.3.1.el8.x86_64-x86_64-with-centos-8.3.2011)\u001b[39;49;00m\n",
      "\u001b[37m   (c) 2016-2022 Aldy Authors. All rights reserved.\n",
      "   Free for non-commercial/academic use only.\u001b[39;49;00m\n",
      "\u001b[37mGenotyping sample MMI011.realigned.bam...\u001b[39;49;00m\n",
      "\u001b[33;01mAverage sample coverage is 19.09512994781937. We recommend at least 20x coverage for the best results.\u001b[39;49;00m\n",
      "\u001b[37mPotential G6PD gene structures for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mPotential major G6PD star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*B (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mBest G6PD star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: *B / *B (confidence=100%)\u001b[39;49;00m\n",
      "\u001b[37m      Minor alleles: *B, *B\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01mG6PD results:\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01m  - *B / *B\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m    Minor: [*B] / [*B]\u001b[39;49;00m\n",
      "\u001b[37m    Legacy notation: [*B (wildtype)] / [*B (wildtype)]\u001b[39;49;00m\n",
      "\u001b[37müêø  Aldy v3.3 (Python 3.7.9 on Linux 4.18.0-305.3.1.el8.x86_64-x86_64-with-centos-8.3.2011)\u001b[39;49;00m\n",
      "\u001b[37m   (c) 2016-2022 Aldy Authors. All rights reserved.\n",
      "   Free for non-commercial/academic use only.\u001b[39;49;00m\n",
      "\u001b[37mGenotyping sample MMI011.realigned.bam...\u001b[39;49;00m\n",
      "\u001b[37mPotential IFNL3 gene structures for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mPotential major IFNL3 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*normal (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mBest IFNL3 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: *normal / *normal (confidence=100%)\u001b[39;49;00m\n",
      "\u001b[37m      Minor alleles: *normal, *normal\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01mIFNL3 results:\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01m  - *normal / *normal\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m    Minor: [*normal] / [*normal]\u001b[39;49;00m\n",
      "\u001b[37m    Legacy notation: [*normal] / [*normal]\u001b[39;49;00m\n",
      "\u001b[37müêø  Aldy v3.3 (Python 3.7.9 on Linux 4.18.0-305.3.1.el8.x86_64-x86_64-with-centos-8.3.2011)\u001b[39;49;00m\n",
      "\u001b[37m   (c) 2016-2022 Aldy Authors. All rights reserved.\n",
      "   Free for non-commercial/academic use only.\u001b[39;49;00m\n",
      "\u001b[37mGenotyping sample MMI011.realigned.bam...\u001b[39;49;00m\n",
      "\u001b[37mPotential NUDT15 gene structures for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mPotential major NUDT15 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mBest NUDT15 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: *1 / *1 (confidence=100%)\u001b[39;49;00m\n",
      "\u001b[37m      Minor alleles: *1.001, *1.001\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01mNUDT15 results:\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01m  - *1 / *1\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m    Minor: [*1.001] / [*1.001]\u001b[39;49;00m\n",
      "\u001b[37m    Legacy notation: [*1.001] / [*1.001]\u001b[39;49;00m\n",
      "\u001b[37müêø  Aldy v3.3 (Python 3.7.9 on Linux 4.18.0-305.3.1.el8.x86_64-x86_64-with-centos-8.3.2011)\u001b[39;49;00m\n",
      "\u001b[37m   (c) 2016-2022 Aldy Authors. All rights reserved.\n",
      "   Free for non-commercial/academic use only.\u001b[39;49;00m\n",
      "\u001b[37mGenotyping sample MMI011.realigned.bam...\u001b[39;49;00m\n",
      "\u001b[37mPotential SLCO1B1 gene structures for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mPotential major SLCO1B1 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 1x*1, 1x*1B & rs4149057, rs2291075 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mBest SLCO1B1 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: *1+rs4149057+rs2291075 / *1B (confidence=100%)\u001b[39;49;00m\n",
      "\u001b[37m      Minor alleles: *1.001 +rs4149057 +rs2291075, *1.002\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01mSLCO1B1 results:\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01m  - *1+rs4149057+rs2291075 / *1B\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m    Minor: [*1.001 +rs4149057 +rs2291075] / [*1.002]\u001b[39;49;00m\n",
      "\u001b[37m    Legacy notation: [*1A +rs4149057 +rs2291075] / [*1B]\u001b[39;49;00m\n",
      "\u001b[33;01mWARNING: mutations rs2291075, rs4149057 suggest presence of a novel major star-allele.\n",
      "However, such alleles cannot be determined without phasing data.\n",
      "Please provide --phase parameter for Aldy to accurately call novel major star-alleles.\n",
      "The above-reported assignments of these mutations are random.\u001b[39;49;00m\n",
      "\u001b[37müêø  Aldy v3.3 (Python 3.7.9 on Linux 4.18.0-305.3.1.el8.x86_64-x86_64-with-centos-8.3.2011)\u001b[39;49;00m\n",
      "\u001b[37m   (c) 2016-2022 Aldy Authors. All rights reserved.\n",
      "   Free for non-commercial/academic use only.\u001b[39;49;00m\n",
      "\u001b[37mGenotyping sample MMI011.realigned.bam...\u001b[39;49;00m\n",
      "\u001b[37mPotential TPMT gene structures for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mPotential major TPMT star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mBest TPMT star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: *1 / *1 (confidence=100%)\u001b[39;49;00m\n",
      "\u001b[37m      Minor alleles: *1.001, *1.001\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01mTPMT results:\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01m  - *1 / *1\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m    Minor: [*1.001] / [*1.001]\u001b[39;49;00m\n",
      "\u001b[37m    Legacy notation: [*1] / [*1]\u001b[39;49;00m\n",
      "\u001b[37müêø  Aldy v3.3 (Python 3.7.9 on Linux 4.18.0-305.3.1.el8.x86_64-x86_64-with-centos-8.3.2011)\u001b[39;49;00m\n",
      "\u001b[37m   (c) 2016-2022 Aldy Authors. All rights reserved.\n",
      "   Free for non-commercial/academic use only.\u001b[39;49;00m\n",
      "\u001b[37mGenotyping sample MMI011.realigned.bam...\u001b[39;49;00m\n",
      "\u001b[37mPotential UGT1A1 gene structures for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mPotential major UGT1A1 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mBest UGT1A1 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: *1 / *1 (confidence=100%)\u001b[39;49;00m\n",
      "\u001b[37m      Minor alleles: *1.001, *1.001\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01mUGT1A1 results:\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01m  - *1 / *1\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m    Minor: [*1.001] / [*1.001]\u001b[39;49;00m\n",
      "\u001b[37m    Legacy notation: [*1] / [*1]\u001b[39;49;00m\n",
      "\u001b[37müêø  Aldy v3.3 (Python 3.7.9 on Linux 4.18.0-305.3.1.el8.x86_64-x86_64-with-centos-8.3.2011)\u001b[39;49;00m\n",
      "\u001b[37m   (c) 2016-2022 Aldy Authors. All rights reserved.\n",
      "   Free for non-commercial/academic use only.\u001b[39;49;00m\n",
      "\u001b[37mGenotyping sample MMI011.realigned.bam...\u001b[39;49;00m\n",
      "\u001b[37mPotential VKORC1 gene structures for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 2x*1 (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mPotential major VKORC1 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: 1x*T, 1x*normal (confidence: 100%)\u001b[39;49;00m\n",
      "\u001b[37mBest VKORC1 star-alleles for MMI011.realigned:\u001b[39;49;00m\n",
      "\u001b[37m   1: *T / *normal (confidence=100%)\u001b[39;49;00m\n",
      "\u001b[37m      Minor alleles: *T, *normal\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01mVKORC1 results:\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m\u001b[32;01m  - *T / *normal\u001b[39;49;00m\u001b[39;49;00m\n",
      "\u001b[37m    Minor: [*T] / [*normal]\u001b[39;49;00m\n",
      "\u001b[37m    Legacy notation: [*T] / [*normal]\u001b[39;49;00m\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/stargazer.py\", line 145, in <module>\n",
      "    main()\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/stargazer.py\", line 133, in main\n",
      "    TOOLS[args.tool].run(args)\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/genotype.py\", line 1305, in run\n",
      "    conformed_vcf = conform_vcf(processed_vcf, ref_vcf) # step 4\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/genotype.py\", line 907, in conform_vcf\n",
      "    fields1[9:] = [f(x) for x in fields1[9:]]\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/genotype.py\", line 907, in <listcomp>\n",
      "    fields1[9:] = [f(x) for x in fields1[9:]]\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/genotype.py\", line 904, in f\n",
      "    ad2[mapping[i]] = ad1[i]\n",
      "KeyError: 3\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/stargazer.py\", line 145, in <module>\n",
      "    main()\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/stargazer.py\", line 133, in main\n",
      "    TOOLS[args.tool].run(args)\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/genotype.py\", line 1305, in run\n",
      "    conformed_vcf = conform_vcf(processed_vcf, ref_vcf) # step 4\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/genotype.py\", line 907, in conform_vcf\n",
      "    fields1[9:] = [f(x) for x in fields1[9:]]\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/genotype.py\", line 907, in <listcomp>\n",
      "    fields1[9:] = [f(x) for x in fields1[9:]]\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/genotype.py\", line 904, in f\n",
      "    ad2[mapping[i]] = ad1[i]\n",
      "KeyError: 3\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/stargazer.py\", line 145, in <module>\n",
      "    main()\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/stargazer.py\", line 133, in main\n",
      "    TOOLS[args.tool].run(args)\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/genotype.py\", line 1305, in run\n",
      "    conformed_vcf = conform_vcf(processed_vcf, ref_vcf) # step 4\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/genotype.py\", line 907, in conform_vcf\n",
      "    fields1[9:] = [f(x) for x in fields1[9:]]\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/genotype.py\", line 907, in <listcomp>\n",
      "    fields1[9:] = [f(x) for x in fields1[9:]]\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/genotype.py\", line 904, in f\n",
      "    ad2[mapping[i]] = ad1[i]\n",
      "KeyError: 3\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/stargazer.py\", line 145, in <module>\n",
      "    main()\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/stargazer.py\", line 133, in main\n",
      "    TOOLS[args.tool].run(args)\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/genotype.py\", line 1305, in run\n",
      "    conformed_vcf = conform_vcf(processed_vcf, ref_vcf) # step 4\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/genotype.py\", line 907, in conform_vcf\n",
      "    fields1[9:] = [f(x) for x in fields1[9:]]\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/genotype.py\", line 907, in <listcomp>\n",
      "    fields1[9:] = [f(x) for x in fields1[9:]]\n",
      "  File \"/home/webb/pharm/pharm_pipe/lib/Stargazer_v1.0.8/genotype.py\", line 904, in f\n",
      "    ad2[mapping[i]] = ad1[i]\n",
      "KeyError: 3\n",
      "MMI012\n",
      "INFO:root:Processing sample MMI012.realigned at 2022-03-02 01:09:12.895708\n",
      "INFO:root:Writing to json at 2022-03-02 01:09:59.231606\n",
      "INFO:root:Writing to tsv at 2022-03-02 01:09:59.232741\n",
      "Call Cyrius to out/MMI012/cyrius_MMI012 done!\n"
     ]
    }
   ],
   "source": [
    "for subject in df.index[21:]:\n",
    "    print(subject)\n",
    "    sex = df.loc[subject, 'gender']\n",
    "    ! python3 star_call.py --id {subject} --out out/{subject} --sex {sex}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "9f0584f11fb420245b0a7c88193370935792d51a09787ed831b17512267b57c2"
  },
  "kernelspec": {
   "display_name": "Python 3.7.9 ('base')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.9"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
